lib/AST/Parameter.cpp

```diff
@@ -64,32 +64,32 @@ ParameterList *ParameterList::clone(const ASTContext &C,
   for (auto &decl : params) {
     auto defaultArgKind = decl->getDefaultArgumentKind();
 
-    decl = ParamDecl::cloneWithoutType(C, decl);
-    if (options & Implicit)
-      decl->setImplicit();
-
-    // If the argument isn't named, give the parameter a name so that
-    // silgen will produce a value for it.
-    if (decl->getName().empty() && (options & NamedArguments)) {
-      llvm::SmallString<16> s;
-      { llvm::raw_svector_ostream(s) << "__argument" << ++i; }
-      decl->setName(C.getIdentifier(s));
-    }
-    
     // If we're inheriting a default argument, mark it as such.
     // FIXME: Figure out how to clone default arguments as well.
     if (options & Inherited) {
       switch (defaultArgKind) {
       case DefaultArgumentKind::Normal:
       case DefaultArgumentKind::StoredProperty:
-        decl->setDefaultArgumentKind(DefaultArgumentKind::Inherited);
+        defaultArgKind = DefaultArgumentKind::Inherited;
         break;
 
       default:
         break;
       }
     } else {
-      decl->setDefaultArgumentKind(DefaultArgumentKind::None);
+      defaultArgKind = DefaultArgumentKind::None;
+    }
+
+    decl = ParamDecl::cloneWithoutType(C, decl, defaultArgKind);
```

Comment:
Currently `options` are consumed only in each `ParamDecl`. It's more like `ParamDecl` clone option than `ParameterList`. I feel like `ParamDecl::cloneWithoutType()` should accept `options` and all the logic in this loop iteration should go in it. WDYT?

Reply:
Makes sense.